nilearn/image/resampling.py

```diff
@@ -474,6 +467,18 @@ def resample_img(img, target_affine=None, target_shape=None,
     else:
         resampled_data_dtype = data.dtype
 
+    # ndimage.affine_transform which is in helper function _resample_one_img
+    # does not handle well data not in native endianness.
+    # Hence, we swap new byte order to change dtype to native endianness.
+    # see https://github.com/nilearn/nilearn/issues/275 &
+    # https://github.com/nilearn/nilearn/issues/1445
+    # This bug is supposed to be fixed in Scipy 0.15 but it is not. Hence we
+    # are not checking any Scipy versions here.
+
+    # Byte swap here, if it is big-endian
+    if not resampled_data_dtype.isnative:
+        resampled_data_dtype = resampled_data_dtype.newbyteorder()
```

Comment:
It does not seem like a great work-around. My guess is that this may not do the right thing with intermediary scipy versions, but I may be wrong about that.

In other words the bug that we reported to scipy was fixed in 0.15 but this one seems slightly different and needs more investigation (I have started digging into it this morning but no completely clear picture right now)